extensions/cba_settings/cba_settings.cpp

```diff
@@ -0,0 +1,335 @@
+
+    string line;
+    while (file.good()) {
+        getline(file, line);
+        line = trim_left(line);
+        string orig = trim_right(line);
+
+        if (line.length() == 0 || result == "true") {
+            if (file.good()) // Don't explode file by appending duplicate additional newlines.
+                temp << orig << endl;
+
+            continue;
+        }
+
+        // Keep commented out lines.
+        if (line.substr(0, 2) == "//") {
+            temp << orig << endl;
+            continue;
+        }
+
+        // Strip force tags.
+        while (line.substr(0, 6) == "force ") {
+            line = line.substr(6);
+            line = trim_left(line);
+        }
+
+        // Get setting.
+        size_t i = line.find("=");
+        if (i == string::npos) // Remove ill-formed line.
+            continue;
+
+        string item = line.substr(0, i);
+        item = trim_right(item);
+
+        if (_stricmp(item.c_str(), setting.c_str()) == 0) {
+            // Upade matching lines.
+            line = "";
+            while (force-- > 0)
+                line = line + "force ";
+
+            line = line + setting + " = " + value + ";";
+
+            temp << line << endl;
+            result = "true";
+        } else {
+            // Keep non-matching lines.
+            temp << orig << endl;
+        }
+    }
+
+    temp.close();
+    file.close();
+
+    if (result == "true") {
+        remove(SETTINGS_FILE);
+        rename(TEMP_FILE, SETTINGS_FILE);
+    } else {
+        remove(TEMP_FILE);
+        result = append(setting, value, force);
+    };
+
+    return result;
+}
+
+extern "C" {
+    __declspec(dllexport) void __stdcall RVExtensionVersion(char* output, int outputSize);
+    __declspec(dllexport) void __stdcall RVExtension(char* output, int outputSize, const char* function);
+    __declspec(dllexport) int __stdcall RVExtensionArgs(char* output, int outputSize, const char* function, const char** args, int argsCnt);
+}
+
+void __stdcall RVExtensionVersion(char* output, int outputSize) {
+    strncpy_s(output, outputSize, VERSION, _TRUNCATE);
+}
+
+void __stdcall RVExtension(char* output, int outputSize, const char* function) {
+    string result = "false";
+
+    if (_stricmp(function, "version") == 0) {
+        result = version();
+        RET;
+    }
+
+    RET;
+}
+
+int __stdcall RVExtensionArgs(char* output, int outputSize, const char* function, const char** args, int argsCnt) {
+    string result = "false";
+
+    if (argsCnt > 3) {
+        RETURN(PARAMS_ERROR_TOO_MANY_ARGS);
+    }
+
+    string setting = "";
+    if (argsCnt >= 1) {
+        setting = args[0];
+        if (setting.substr(0, 1) == "\"") // Strip extra quote marks.
+            setting = setting.substr(1, setting.length() - 2);
+    }
+
+    string value = "";
+    if (argsCnt >= 2)
+        value = args[1];
+
+    int force = 0;
+    if (argsCnt >= 3)
+        force = min(stoi(args[2]), 2);
+
+    if (_stricmp(function, "version") == 0) {
```

Comment:
why even check the whole string?
c
l
p
r
a

the first character would be sufficient for what you need

Reply:
True I guess, but what does it matter? This is not something that needs to be particularly fast.

Reply:
Doesn't really matter, just my "every nanosecond counts" bullshit.

Reply:
I understand.